components/config-model-migrator/org.wso2.is.configuration.diff.creator/src/main/java/org/wso2/is/configuration/diff/creator/ConfigLoader.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.is.configuration.diff.creator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.is.configuration.diff.creator.exception.ConfigMigrateException;
+import org.wso2.is.configuration.diff.creator.utils.MigrationConstants;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Class for loading config files.
+ */
+public class ConfigLoader {
+
+    private Map<String, File> defaultXMLFiles = new HashMap<>();
+    private Map<String, File> migratedXMLFiles = new HashMap<>();
+    private Map<String, File> defaultPropertiesFiles = new HashMap<>();
+    private Map<String, File> migratedPropertiesFiles = new HashMap<>();
+    private Map<String, File> j2TemplateFiles = new HashMap<>();
+    private static final int TAG_KEY_COLUMN_NO_INDEX = 2;
+
+    private static final Logger log = LogManager.getLogger(ConfigLoader.class);
+
+    /**
+     * Validate whether file path is blank and exists.
+     *
+     * @param migratedISHomePath Migrated IS home path.
+     * @param defaultISHomePath  Default IS home path.
+     */
+    public void validate(String migratedISHomePath, String defaultISHomePath) throws ConfigMigrateException {
+
+        String errorMsg;
+        if (StringUtils.isBlank(migratedISHomePath) && StringUtils.isBlank(defaultISHomePath)) {
+            errorMsg = "Please enter migrated IS home path and default IS home path as inputs.";
+            log.error(errorMsg);
+            throw new ConfigMigrateException(errorMsg);
+        }
+        if (!new File(migratedISHomePath).exists() && !new File(defaultISHomePath).exists()) {
+            errorMsg = "The provided IS home paths does not exist. Please try again after entering correct input" +
+                    " Paths.";
+            log.error(errorMsg);
```

Comment:
What is the reason of logging and throwing the same error message?

Reply:
I need to break the flow. To stop from here without proceeding so used exception the same message